x-pack/platform/plugins/shared/fleet/server/types/models/download_sources.ts

```diff
@@ -23,6 +30,18 @@ const DownloadSourceBaseSchema = {
       }),
     ])
   ),
+  ssl: schema.maybe(
+    schema.object({
+      certificate_authorities: schema.maybe(schema.arrayOf(schema.string())),
+      certificate: schema.maybe(schema.string()),
+      key: schema.maybe(schema.string()),
```

Comment:
should we allow null here?

Reply:
Good question, I was going with the same basic schema as the other `ssl.key` values that I added recently. neither [fleet server host](https://github.com/elastic/kibana/blob/e0bf98e45a6c2b4e138cac601c6082ca68fb8301/x-pack/platform/plugins/shared/fleet/server/types/rest_spec/fleet_server_policy_config.ts#L28) nor [outputs](https://github.com/elastic/kibana/blob/e0bf98e45a6c2b4e138cac601c6082ca68fb8301/x-pack/platform/plugins/shared/fleet/server/types/models/output.ts#L78) have it nullable., but maybe they should all be nullable instead? 

I think it makes sense to keep their behavior aligned as they work the same way.

Reply:
maybe we don't need it, tested locally and works fine when updating a plain text ssl.key to a secret